src/Okta.Idx.Sdk/IdxClient.cs

```diff
@@ -1310,25 +1318,35 @@ private async Task<AuthenticationResponse> EnrollAuthenticatorAsync(IdxRequestPa
             {
                 currentRemediationType = RemediationType.EnrollAuthenticator;
             }
+            else if (selectAuthenticatorResponse.ContainsRemediationOption(RemediationType.EnrollPoll))
+            {
+                currentRemediationType = RemediationType.EnrollPoll;
+            }
 
-            if (currentRemediationType != RemediationType.EnrollAuthenticator &&
-                    currentRemediationType != RemediationType.AuthenticatorEnrollmentData)
+            if (currentRemediationType != RemediationType.EnrollPoll &&
+            currentRemediationType != RemediationType.EnrollAuthenticator &&
+            currentRemediationType != RemediationType.AuthenticatorEnrollmentData)
             {
                 throw new UnexpectedRemediationException(
                     new List<string>
                     {
+                            RemediationType.EnrollPoll,
                             RemediationType.AuthenticatorEnrollmentData,
                             RemediationType.EnrollAuthenticator,
                     },
                     selectAuthenticatorResponse);
             }
 
-            return new AuthenticationResponse
+            var authenticationResponse = new AuthenticationResponse
             {
                 IdxContext = idxContext,
                 AuthenticationStatus = status,
                 CurrentAuthenticator = IdxResponseHelper.ConvertToAuthenticator(selectAuthenticatorResponse.Authenticators.Value, selectAuthenticatorResponse.CurrentAuthenticator.Value, selectAuthenticatorResponse.AuthenticatorEnrollments.Value),
             };
+
+            authenticationResponse.OktaVerifyEnrollOptions = new OktaVerifyEnrollOptions(authenticationResponse, selectAuthenticatorResponse);

```

Comment:
We should follow the current design where all the information of a specific authenticator is accessed via either CurrentAuthenticator or CurrentAuthenticatorEnrollment.